ocaml/xapi/xapi_pusb.ml

```diff
@@ -47,11 +47,24 @@ let scan_start ~__context usbs =
   (* Create the newly added pusbs *)
   USBSet.iter (fun s -> let self = create ~__context ~uSB_group:(Ref.null) ~host ~other_config:[] ~path:s.USB.path ~vendor_id:s.USB.vendor_id
                    ~vendor_desc:s.USB.vendor_desc ~product_id:s.USB.product_id ~product_desc:s.USB.product_desc ~serial:s.USB.serial
-                   ~version:s.USB.version ~description:s.USB.description in
+                   ~version:s.USB.version ~description:s.USB.description ~speed:s.USB.speed in
                let group = Xapi_pusb_helpers.find_or_create ~__context self in
                Db.PUSB.set_USB_group ~__context ~self ~value:group
   ) (USBSet.diff local_usb_set known_usb_set);
 
+  let refresh_speed (self, rec') =
+    if rec'.API.pUSB_speed = "" then begin
+      match USBSet.find_first_opt (fun elt -> USB.compare elt (extract_known_usb_info rec') = 0) local_usb_set with
+      | None -> ()
+      | Some x -> Db.PUSB.set_speed ~__context ~self ~value:x.USB.speed
+    end
+  in
+
+  (* Speed was added to the usb datamodel, so we need
+     to update that field for any usbs that were plugged
+     in before that update. *)
+  List.iter refresh_speed known_pusbs_in_db;
```

Comment:
ideally I would simply replace the existing records completely so we are robust to datamodel changes, but I think that might cause problems for any USB that is plugged (see try/catch below)